core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -36,6 +37,31 @@ trait Runtime[+R] {
    */
   val platform: Platform
 
+  /**
+   * Add generic signal handler
+   *
+   * @param signal to listen for
+   * @param handler to run on receiving signal
+   */
+  def addSignalHandler(signal: String)(handler: URIO[R, Unit]): Unit =
```

Comment:
I'm wondering if we should move these methods to `Platform` and then provide aliases in `Runtime`, similar to the existing treatment of methods like `withReportFailure`.